crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -451,6 +457,28 @@ def delete_pacemaker_resources(node_name)
         end
       end
 
+      # Evacuate all routers away from the specified network node to other
+      # available network nodes. The evacuation procedure it started on the
+      # specified controller node
+      def evacuate_network_node(controller, network_node, delete_namespaces = false)
+        args = [network_node]
+        args << "delete-ns" if delete_namespaces
+        controller.wait_for_script_to_finish(
+          "/usr/sbin/crowbar-router-migration.sh", 600, args
```

Comment:
Yeah, I am not really happy with that either. And 10 minutes is clearly not enough for a somewhat bigger setup.
We will need to rethink a few other timeouts here as well.